### PR TITLE
chore(pe): context duplication noodling InstrumentContext

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

A PR into the noodling branch. Primarily this is about removing the contents of the duplicated `InstrumentContext`, `Well` and `Labware`.

# Changelog

- remove all private methods from `InstrumentContext`, `Labware` and `Well`.
- replace implementations of all public methods in `InstrumentContext`, `Labware` and `Well`. with `raise NotImplementedError()`

# Review requests

Should I also remove docstrings? 

# Risk assessment

None

### CODE REVIEW

File: api/.flake8 [hunk 1]
```diff
@@ -36,6 +36,7 @@ per-file-ignores =
     src/opentrons/data_storage/*:ANN,D
     src/opentrons/deck_calibration/*:ANN,D
     src/opentrons/drivers/*:ANN,D
+    src/opentrons/experimental_protocol_api/*:ANN,D
```
Comment (mcous) [inline api/.flake8]: Let's not disable these lints in our new module

File: api/src/opentrons/experimental_protocol_api/instrument_context.py [hunk 2]
```diff
@@ -25,59 +28,27 @@ class InstrumentContext(CommandPublisher):
...
-            self.trash_container = self._ctx.fixed_trash
-        else:
-            self.trash_container = trash
-
-        self._last_tip_picked_up_from: Union[Well, None] = None
-        self._starting_tip: Union[Well, None] = None
-        self.requested_as = self._implementation.get_instrument_name()
-
     @property  # type: ignore
     @requires_version(2, 0)
```
Comment (mcous) [inline api/src/opentrons/experimental_protocol_api/instrument_context.py]: Is `requires_version` functionality we should leave out for now?

File: api/src/opentrons/experimental_protocol_api/instrument_context.py [hunk 3]
```diff
@@ -25,59 +28,27 @@ class InstrumentContext(CommandPublisher):
...
-        if trash is None:
-            self.trash_container = self._ctx.fixed_trash
-        else:
-            self.trash_container = trash
-
-        self._last_tip_picked_up_from: Union[Well, None] = None
-        self._starting_tip: Union[Well, None] = None
-        self.requested_as = self._implementation.get_instrument_name()
-
     @property  # type: ignore
```
Comment (mcous) [inline api/src/opentrons/experimental_protocol_api/instrument_context.py]: What's the `type: ignore` for?

File: api/src/opentrons/experimental_protocol_api/instrument_context.py [hunk 4]
```diff
@@ -963,95 +601,15 @@ def transfer(self,
               with the `gradient` keyword argument to create a custom curve.
 
         :returns: This instance
+
+        Args:
```
Comment (mcous) [inline api/src/opentrons/experimental_protocol_api/instrument_context.py]: Inconsistent docstrings. Crazy thought: should we leave docstrings out of this for now? That way, we _must_ go through and re-write all of them (or at least give the existing ones a once-over before moving them over)

Comment (amitlissack) [reply api/src/opentrons/experimental_protocol_api/instrument_context.py]: That was my question in the PR. :)

I dig it.

File: api/src/opentrons/experimental_protocol_api/instrument_context.py (hunk 3 above)
Comment (amitlissack) [reply api/src/opentrons/experimental_protocol_api/instrument_context.py]: I don't know. I'll check it out.

File: api/src/opentrons/experimental_protocol_api/instrument_context.py (hunk 2 above)
Comment (amitlissack) [reply api/src/opentrons/experimental_protocol_api/instrument_context.py]: Sure.

File: api/.flake8 (hunk 1 above)
Comment (amitlissack) [reply api/.flake8]: I will keep it disabled if we keep the docstrings.

File: api/src/opentrons/experimental_protocol_api/instrument_context.py (hunk 4 above)
Comment (mcous) [reply api/src/opentrons/experimental_protocol_api/instrument_context.py]: 🤦 that's what I get for not taking my time and reading the PR description.

But yeah, my vote is yes!

Comment (SyntaxColoring) [reply api/src/opentrons/experimental_protocol_api/instrument_context.py]: I agree with removing the docstrings. We should document the places where our behavior might differ from the old, but in general, we should defer to the existing contexts' reference docs.

File: api/src/opentrons/experimental_protocol_api/instrument_context.py [hunk 5]
```diff
@@ -1,1397 +1,261 @@
...
+from typing import Optional, Union, Sequence, List
 
+from opentrons import APIVersion, types
+from opentrons.experimental_protocol_api.labware import Well, Labware
+from opentrons.hardware_control.dev_types import PipetteDict
+from opentrons.protocol_api import PairedInstrumentContext
+from opentrons.protocol_api.instrument_context import AdvancedLiquidHandling
+from opentrons.protocols.api_support.util import (
+    PlungerSpeeds, FlowRates, Clearances
+)
```
Comment (SyntaxColoring) [inline api/src/opentrons/experimental_protocol_api/instrument_context.py:13]: Not a problem for this PR, but most of these imports look like red flags to me.

For example, `opentrons.APIVersion` seems like another example of the `opentrons.types.Point` problem. It's an internal type that doubles as part of the top-level interface—which expands and confuses the set of namespaces that a user of the top-level interface has to care about.

File: api/src/opentrons/experimental_protocol_api/labware.py [hunk 6]
```diff
@@ -1,946 +1,241 @@
...
-                f'version or update your robot.')
-        self._api_version = api_level
-        self._implementation = implementation
+    # noqa: D101
 
     @property
     def separate_calibration(self) -> bool:
-        return self._implementation.separate_calibration
+        # noqa: D102
+        raise NotImplementedError()
```
Comment (SyntaxColoring) [inline api/src/opentrons/experimental_protocol_api/labware.py:127]: Doesn't have to happen in this PR, but, for:

* Non-underscore-prefixed methods that aren't explained on docs.opentrons.com
* Non-underscore-prefixed methods that *are* explained on docs.opentrons.com, but so vaguely that no user could ever reasonably use them (like this `LabwareContext.separate_calibration`, or `LabwareContext.quirks`)

I think we should:

* Strip them out of these files, as if they were underscore-prefixed and we were removing private methods.
* Leave a note somewhere to remind us that we removed them.
* Later, figure out if we need to add them back in—and, if so, how.

**Review by SyntaxColoring — APPROVED** [review]
Nice!

**Pull request merged**